File: server/src/models/Donation.js
/**
 * Donation Model - Donasi dari user
 * File: src/models/Donation.js
 */

const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Donation = sequelize.define('Donation', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  amount: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false,
    comment: 'Jumlah donasi'
  },
  method: {
    type: DataTypes.ENUM('qris', 'gopay', 'ovo', 'dana', 'shopeepay', 'bca', 'bni', 'mandiri', 'other'),
    allowNull: false
  },
  proof_url: {
    type: DataTypes.STRING(500),
    allowNull: true,
    comment: 'URL bukti transfer'
  },
  status: {
    type: DataTypes.ENUM('pending', 'verified', 'rejected'),
    defaultValue: 'pending'
  },
  donor_name: {
    type: DataTypes.STRING(255),
    allowNull: true,
    comment: 'Nama pengirim (opsional)'
  },
  message: {
    type: DataTypes.TEXT,
    allowNull: true,
    comment: 'Pesan dari donatur'
  },
  admin_notes: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  verified_by: {
    type: DataTypes.INTEGER,
    allowNull: true,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  verified_at: {
    type: DataTypes.DATE,
    allowNull: true
  }
}, {
  tableName: 'donations',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at'
});

module.exports = Donation;

File: native/android/production/karaoke_native/features/donations/models/donation.dart
import 'package:equatable/equatable.dart';

class Donation extends Equatable {
  final int id;
  final double amount;
  final String method;
  final String? proofUrl;
  final String status;
  final String? donorName;
  final String? message;
  final String? adminNotes;

  const Donation({
    required this.id,
    required this.amount,
    required this.method,
    this.proofUrl,
    required this.status,
    this.donorName,
    this.message,
    this.adminNotes,
  });

  factory Donation.fromJson(Map<String, dynamic> json) {
    return Donation(
      id: _parseInt(json['id']),
      amount: double.tryParse(json['amount']?.toString() ?? '0') ?? 0,
      method: json['method']?.toString() ?? 'other',
      proofUrl: json['proof_url']?.toString(),
      status: json['status']?.toString() ?? 'pending',
      donorName: json['donor_name']?.toString(),
      message: json['message']?.toString(),
      adminNotes: json['admin_notes']?.toString(),
    );
  }

  @override
  List<Object?> get props => [id, amount, method, proofUrl, status, donorName, message, adminNotes];

  static int _parseInt(dynamic value) {
    if (value == null) return 0;
    if (value is int) return value;
    return int.tryParse(value.toString()) ?? 0;
  }
}
